crates/bifrost/src/providers/replicated_loglet/tasks/get_trim_point.rs

```diff
@@ -78,6 +78,14 @@ impl<'a> GetTrimPointTask<'a> {
             .nodeset
             .to_effective(&networking.metadata().nodes_config_ref());
 
+        if effective_nodeset.is_empty() {
+            // all nodes are disabled, we can't determine the trim point but we know that this
+            // loglet is impossible to read. Control plane transitioning our nodeset into disabled
+            // must have trimmed all records. In all cases, it's safe to return the maximum
+            // possible trim point.
+            return Ok(Some(LogletOffset::MAX.prev()));
+        }
```

Comment:
It's ok to return MAX here because we clamp it back to the `tail_lsn` in `LogletWrapper` and it must have been sealed before the corresponding log servers could have been disabled, right?

Reply:
Good point